src/main/java/org/prebid/server/auction/AuctionRequestFactory.java

```diff
@@ -277,16 +281,41 @@ private Device populateDevice(Device device, HttpServerRequest request) {
 
         if (StringUtils.isBlank(ip) || StringUtils.isBlank(ua)) {
             final Device.DeviceBuilder builder = device == null ? Device.builder() : device.toBuilder();
-            builder.ip(StringUtils.isNotBlank(ip) ? ip : paramsExtractor.ipFrom(request));
             builder.ua(StringUtils.isNotBlank(ua) ? ua : paramsExtractor.uaFrom(request));
-
+            final String ipFromRequest = paramsExtractor.ipFrom(request);
+            final InetAddress inetAddress = inetAddressByIp(ipFromRequest);
+            if (isIpv4(inetAddress, ipFromRequest)) {
+                builder.ip(ipFromRequest);
+            } else if (isIpv6(inetAddress)) {
+                builder.ipv6(ipFromRequest);
+            }
```

Comment:
When you adding something in tests, you need to clearly understand why you did that. 
You forgot about 
```
if StringUtils.isBlank(ip)
```